base/compiler/tfuncs.jl

```diff
@@ -302,16 +302,30 @@ function sizeof_nothrow(@nospecialize(x))
         end
     elseif isa(x, Conditional)
         return true
-    else
-        x = widenconst(x)
     end
     if isa(x, Union)
         return sizeof_nothrow(x.a) && sizeof_nothrow(x.b)
     end
-    isconstType(x) && (x = x.parameters[1]) # since sizeof(typeof(x)) == sizeof(x)
-    x === DataType && return false
+    t, exact = instanceof_tfunc(x)
+    if !exact
+        # Could always be bottom at runtime, which throws
+        return false
+    end
+    if t !== Bottom
+        # The value corresponding to `x` at runtime could be a type.
```

Comment:
I think this still gives the wrong answer for e.g. `sizeof_nothrow(Type{Vector{Int}})` (or `Type{String}`). `sizeof(Vector{Int})` throws.

Reply:
Good catch.